crates/router/src/connector/klarna/transformers.rs

```diff
@@ -49,9 +49,12 @@ impl TryFrom<&types::PaymentsSessionRouterData> for KlarnaSessionRequest {
                 order_amount: request.amount,
                 locale: "en-US".to_string(),
                 order_lines: vec![OrderLines {
-                    name: order_details.product_name,
-                    quantity: order_details.quantity,
-                    unit_price: request.amount,
+                    name: order_details
+                        .iter()
+                        .map(|v| v.product_name.clone())
+                        .collect(),
+                    quantity: order_details.iter().map(|v| v.quantity).collect(),
+                    unit_price: order_details.iter().map(|v| v.amount).collect(),
```

Comment:
Should the vector be populated with each order_detail ? Right now I guess it would populate it as `[{"name":["name1", "name2"]}]` .

cc: @Narayanbhat166

Reply:
Yes, you are right @Abhicodes-crypto. According to their [docs](https://docs.klarna.com/api/payments/#operation/createOrder!path=order_lines&t=request) this should be the vector of order lines.